spec/unit/knife/winrm_session_spec.rb

```diff
@@ -0,0 +1,99 @@
+#
+# Author:: Steven Murawski <smurawski@chef.io>
+# Copyright:: Copyright (c) 2015-2020 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "spec_helper"
+require_relative "../../../lib/chef/knife/winrm"
+require_relative "dummy_winrm_connection"
+
+describe Chef::Knife::WinrmSession do
+  let(:winrm_connection) { Dummy::Connection.new }
+  let(:options) { { transport: :plaintext } }
+
+  before do
+    @original_config = Chef::Config.hash_dup
+    allow(WinRM::Connection).to receive(:new).and_return(winrm_connection)
+  end
+
+  after do
+    Chef::Config.configuration = @original_config
+  end
```

Comment:
```suggestion
  before do
    allow(WinRM::Connection).to receive(:new).and_return(winrm_connection)
  end
```

This gets reset before every test.